### PR TITLE
C param for ZProbe Mode 4

### PR DESCRIPTION
**Pull request by andreq**
Add param so user can choose which Endstop switch to use as a Zprobe input for Mode 4.
Deprecated Mode 6 (fallback to Mode 4 with E1 switch)

### CODE REVIEW

File: src/ZProbe.cpp [hunk 1]
```diff
@@ -24,6 +24,7 @@ void ZProbe::Init(float h)
 	tolerance = DefaultZProbeTolerance;
 	maxTaps = DefaultZProbeTaps;
 	invertReading = turnHeatersOff = false;
+	eEndstop = 1; // Default e1 for Mode 6 backward compatibility
```
Comment (andreq) [inline src/ZProbe.cpp]: I just realized that this would "break" default Mode 4 behavior.
I have no quick fix for a nicer way to handle that right now.

**Comment by dc42** [comment]
I plan to included it in 2.02, however it is now tied in with Duet 3 work so the details of the parameter may change.

**Comment by andreq** [comment]
Should I close the PR or you wanna do it? Looks like you re-wrote the code anyway.

**Comment by dc42** [comment]
Implemented with changes in version 2.02RC3. Thanks for your contribution!

**Pull request closed without merging**